src/goto-programs/remove_exceptions.cpp

```diff
@@ -98,175 +101,114 @@ class remove_exceptionst
   std::map<irep_idt, std::set<irep_idt>> &exceptions_map;
   bool remove_added_instanceof;
 
-  void add_exceptional_returns(
-    const irep_idt &function_id,
-    goto_programt &goto_program);
+  symbol_exprt get_inflight_exception_global();
+
+  bool may_throw_or_receive_exceptions(const goto_programt &) const;
 
   void instrument_exception_handler(
-    const irep_idt &function_id,
     goto_programt &goto_program,
-    const goto_programt::targett &);
+    const goto_programt::targett &,
+    bool may_catch);
 
   void add_exception_dispatch_sequence(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &instr_it,
     const stack_catcht &stack_catch,
     const std::vector<exprt> &locals);
 
   void instrument_throw(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &,
     const stack_catcht &,
     const std::vector<exprt> &);
 
   void instrument_function_call(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &,
     const stack_catcht &,
     const std::vector<exprt> &);
 
   void instrument_exceptions(
-    const irep_idt &function_id,
     goto_programt &goto_program);
 };
 
-/// adds exceptional return variables for every function that may escape
-/// exceptions
-void remove_exceptionst::add_exceptional_returns(
-  const irep_idt &function_id,
-  goto_programt &goto_program)
+/// Create a global named java::@inflight_exception that holds any exception
```

Comment:
I think doxygen fails on this use of `@`, unless I'm misreading the error message.